k9mail/src/main/java/com/fsck/k9/notification/AuthenticationErrorNotifications.java

```diff
@@ -44,6 +46,10 @@ public void showAuthenticationErrorNotification(Account account, boolean incomin
                 .setStyle(new BigTextStyle().bigText(text))
                 .setVisibility(NotificationCompat.VISIBILITY_PUBLIC);
 
+        if (Build.VERSION.SDK_INT >= 21) {
```

Comment:
`NotificationCompat` exists precisely to avoid this kind of conditional code.

We use `NotificationCompat.Builder` everywhere. So simply calling `setCategory()` on it should do the right thing.